osu.Game.Tests/NonVisual/Filtering/FilterQueryParserTest.cs

```diff
@@ -316,5 +316,86 @@ public bool TryParseCustomKeywordCriteria(string key, Operator op, string value)
                 return false;
             }
         }
+
+        //Date criteria testing
+
+        private static readonly object[] correct_date_query_examples =
+        {
+            new object[] { "600" },
+            new object[] { "0.5s" },
+            new object[] { "120m" },
+            new object[] { "48h120s" },
+            new object[] { "10y24M" },
+            new object[] { "10y60d120s" },
+            new object[] { "0.1y0.1M2d" },
+            new object[] { "0.99y0.99M2d" }
+        };
+
+        [Test]
+        [TestCaseSource(nameof(correct_date_query_examples))]
+        public void TestValidDateQueries(string dateQuery)
+        {
+            string query = $"played={dateQuery} time";
```

Comment:
Equality filter does not make sense in this context I agree. It could be removed or used to implement some kind of lenience like for example: `played=0` would show only the beatmaps played today. But this would be different than stable behavior.

Reply:
I think it's best to remove it. It's unclear what magnitude of tolerance you would allow for an equality filter.

Reply:
` equal` as been removed in all dates testing. I have a follow up question on that. How should be the behavior of the criteria if the user inputs for example `played=5`? Should the filter be created or not? The visual difference for the user would be:

| Filter created (currently) | Filter not created |
|-|-|
| ![osu_2023-04-05_12-26-59](https://user-images.githubusercontent.com/39671719/230054729-b550c0d1-a153-4a85-829e-07411e6f5353.jpg) | ![osu_2023-04-05_12-23-44](https://user-images.githubusercontent.com/39671719/230054012-9fdeb9d7-ea39-4759-828c-aaa1ab554af4.jpg) |

Reply:
Both options kinda suck. But the second probably sucks less (it's *some* indicaton that the filter didn't parse as a date filter).

Reply:
I implemented the second option. I can't find anything better with the current tools. I don't think there is way to warn the user that a filter is not "correctly" used.